cloud/amazon/ec2_group_facts.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_group_facts
+author: "Stefan Horning (@stefanhorning)"
+version_added: "2.2"
+short_description: Finds exising EC2 security groups and returns their facts
+description:
+    - Finds a list of existing EC2 security groups based on given filters. Returns the facts of each found security group. This module has a dependency on boto3.
+options:
+  name:
+    description:
+      - Name of the security group.
+    required: false
+  group_id:
+    description:
+      - ID of the security group.
+    required: false
+  description:
+    description:
+      - Description of the security group.
+    required: false
+  vpc_id:
+    description:
+      - ID of the VPC the security belongs to.
+    required: false
```

Comment:
Isn't it more user friendly this way, after all we want to abstract away from AWS internals. I would understand the filter approach if it would be huge set of arbitrary filters. But in this case it's just this concise list of search params.